rules/framework/vfs_overlay.bzl

```diff
@@ -376,6 +376,12 @@ def make_vfsoverlay(ctx, hdrs, module_map, private_hdrs, has_swift, swiftmodules
     vfs_parent_len = len(vfs_parent.split("/")) - 1
     vfs_prefix = _make_relative_prefix(vfs_parent_len)
 
+    # Ensures the determinism of vfsoverlay file
+    # hdrs, private_hdrs and swiftmodules of the same framework may be in different order.
+    hdrs = sorted(hdrs)
```

Comment:
I'd suggest to verify that these sort steps don't introduce a significant performance regression in the analysis phase.

Reply:
The overhead should be fine. [apple_library](https://github.com/bazel-ios/rules_ios/blob/master/rules/library.bzl#L527) has already sorted all srcs files. The number of hdrs, private_hdrs and swiftmodules should be smaller than that, especially for swift modules.

Reply:
I wonder what kind of CI test can red green on this... sandbox mode only ensure all inputs that should be present are present, but nothing about ensuring no change on file content when a rebuild happend

Reply:
maybe start with adding before after in the PR desc? I am sure we can figure some CI test out in future based on the repro